### DISCUSSION TITLE
Might be of interest : docker images + shell version of pgtune

### DISCUSSION THREAD

**Discussion by esgn** [le0pard/pgtune#59] [Show and tell]
Hi all,

I regularly have to deploy APIs and services using PostgreSQL/PostGIS on different server instances (cloud, dedicated, etc) for different use-cases (teaching, testing, prototyping, etc). 
I took the habit of using https://pgtune.leopard.in.ua/ (thanks a lot for it !) to improve PostgreSQL default configuration. However gathering the server system specs, going to pgtune website, copy-pasting options, applying them to my PostgreSQL was a bit cumbersome.
I may have missed something, but as I couldn't find any REST APIs for pgtune or other way to use it without going through the website, I resorted to do this : [this is not a place for free advertisement]
In short :
- I adapted the pgtune code of le0pard/pgtune to make a shell script out of it and also added basic _automatic detection_ of some system settings (ram, cpu count, etc)
- I built PostgreSQL/PostGIS docker images based on the official Docker postgres image and making use of the shell version of pgtune
I'm still in the process of testing the resulting images in different projects. Will be interested by any comments, thoughts or ideas regarding this attempt.

Cheers and thanks for all the work.

**Comment by le0pard** [comment]
For this exists original https://github.com/gregs1104/pgtune
